### PR TITLE
Create `parser` package. Move logic from `xpk.py` to `parser` package. 

### PR DESCRIPTION
**Pull request by PBundyra**
## Fixes / Features
- Create `parser` package
- Initialize `core` package
- Move logic from `xpk.py` to `parser` package. 
- Wrap some of the parser logic into new functions e.g. `set_parser`
- Adjust `pyproject.toml` accordingly

## Testing / Documentation
Tested manually

- [ y ] Tests pass
- [ n ] Appropriate changes to documentation are included in the PR - no user-facing changes

### CODE REVIEW

**Comment by PBundyra** [comment]
Once I'll get LGTM I'll merge it when #166 is merged

File: .github/workflows/build_tests.yaml
Comment (Obliviour) [inline .github/workflows/build_tests.yaml:56]: Oh perfect I was looking for this in the previous PR. I think we want to test both flows with python3 xpk.py and xpk (through pip)?

Maybe a build test that checks that [xpk.py / xpk] --help can be run is a good check for both flows in the build test, the rest of the build test can use just one of them. We could add a nightly test that tests our exhaustive command list?

File: .github/workflows/build_tests.yaml
Comment (Obliviour) [inline .github/workflows/build_tests.yaml:66]: I propose we keep the name xpk.py as the entry point, a handful of our documentation and user workflows is build upon xpk.py and would be a large change from calling python3 xpk.py to python3 main.py.

**Review by Obliviour — COMMENTED** [review]
Awesome refactor, hoping we can keep the name as xpk.py for user experience.

File: .github/workflows/build_tests.yaml
Comment (PBundyra) [reply .github/workflows/build_tests.yaml:66]: I see. At first I also wanted to keep it as `xpk.py` but I encountered some limitation due to python import system. A package cannot have the same name as module within it(in this case `xpk` and `xpk.py`) - see [second answer from this Stackoverflow post](https://stackoverflow.com/questions/54333865/python-no-module-named-error-package-is-not-a-package). At least, this is what happens when running xpk directly using python interpreter. It works correctly when installing first with pip, and then running it as a command. To solve this we have a few alternatives:
### Introduce `main.py` with drawbacks you mentioned
I agree that it's important to be consistent with our documentation 
### Recommend always installing xpk with pip first and discourage running it directly using python interpreter
I like this approach the most. I believe it's a standard practice to use it that way, and it's what other tools do. So from users' perspective, they would need to use it following `README.md`
1. Run `pip install xpk`
2. Run `xpk [subcommand]` 

However, I may not be aware of some cons you are - maybe it would break some customers' workflows?

Another advantage of this approach is that it would be easier to maintain the project, because of simplified e2e tests - we would need to cover only one instead of two ways of installing XPK

### Change name of the package `xpk`
Cons: I believe it's common practice to name packages the same as the project in Python, so we wouldn't be following the convention.


Please let me know what you think

File: .github/workflows/build_tests.yaml
Comment (PBundyra) [reply .github/workflows/build_tests.yaml:56]: Please see my comment below. I believe it's easier to maintain just xpk through pip, and stop supporting using it through python interpreter - unless there are some cons I'm not aware of (maybe some users already use it with the python interpreter and it would break their workflows(?))

File: .github/workflows/build_tests.yaml
Comment (Obliviour) [reply .github/workflows/build_tests.yaml:66]: We have most users using xpk through python3 xpk.py so changing that flow would require a lot of user awareness :( 

No good answer currently

File: xpk/core/core.py
Comment (PBundyra) [reply xpk/core/core.py:1]: >> Do you think if we adjust some naing of the files/folders we can eliminate confusion? There are two files and a folder/package with the name of "core".

Looking at different Python projects (e.g. [JAX](https://github.com/google/jax/tree/main/jax)) I believe it's fine to have two (or even more) files/package with the same name. 

>> I'm new to python but is it a good practice to have a separate module for the only core.py?

You're absolutely right. This is only temp solution until I divide `core.py` into smaller modules. I didn't want to make a huge refactor at once because it would be harder to review and iterate. My goal is to have modules for at least `cluster.py`, `workload.py` and `inspector.py` - similarly to `parser` package, maybe even more, and a separate file/folder for the constants.

File: .github/workflows/build_tests.yaml
Comment (PBundyra) [reply .github/workflows/build_tests.yaml:66]: I see, let me investigate a little further then

Comment (PBundyra) [reply .github/workflows/build_tests.yaml:66]: I came up with [the solution](https://github.com/google/xpk/pull/166), PTAL

Comment (Obliviour) [reply .github/workflows/build_tests.yaml]: Thanks @PBundyra LGTM

File: xpk/core/core.py
Comment (Obliviour) [reply xpk/core/core.py:1]: Can even have a file for the template jobset / kueue configs.

There may be some common core code between cluster.py, workload.py and inspector.py as well. Inspector logic reuses logic from workload list for example.

**Review by Obliviour — APPROVED** [review]
Thank you for the refactor changes, nice way to start work in XPK!

**Pull request merged**
